### PR TITLE
Adds the ability to see pda messages in Deadchat

### PR DESCRIPTION
**Pull request by Mandurrrh**
Port from bay. Gives ghosts/observers the ability to see pda messages when GhostPDA is toggled on.

Edit: Now has its own toggle.

### CODE REVIEW

**Comment by dannno** [comment]
why?

**Comment by Fox-McCloud** [comment]
:-1: 

Sometimes PDAs are used for legitimate private conversation, such as exchanging BYOND info, and the likes; while I think it's a good thing that admins can see these messages (or investigate the PDA log), generally speaking, giving this option to everyone is kinda invasive, in my opinion.

Yes, I realize that you can use the message monitor, but this is fairly limited and isolated.

**Comment by xxalpha** [comment]
>sometimes PDAs are used for legitimate private conversation

All PDA messages are logged in the tcomms computers, that's really terrible if people do that.

**Comment by Antimattercarp** [comment]
:+1: Legit Private communication? What?

**Comment by Chocobro** [comment]
pda messages aren't used for ooc comms - I don't see why they should be. if you're that desperate for someone's byond info, ahelp and ask the admins for it or wait for the round end and ask who Fuckhead Bozo was.

:+1: to this.

**Comment by Dorsisdwarf** [comment]
Yeah, the most OOC thing you should be PDAing is links to the wiki/youtubes really.

**Comment by Mandurrrh** [comment]
I wanna see malfai/onehumanai be creepy or set up plans. Plus funny pda messages. Irc on your pda is not displayed so you could send any private info there? Or just ask who they are in ooc after...

**Comment by TheVekter** [comment]
OOC comms should be happening in B

**Comment by Chocobro** [comment]
or in the ooc channel.

**Comment by GunHog** [comment]
I miss having this as dead AI! (You still get them, but you cannot look around as a dead AI any more). Btw, PDA messages should not be used for OOC things - PDA messages are not private, as they can be accessed by the CE, emag users, and Traitor/Malfunctioning AIs.

**Comment by Dorsisdwarf** [comment]
Doesn't the AI occasionally intercept PDA messages automatically?

**Comment by GunHog** [comment]
No, that feature was removed, sadly.

**Comment by optimumtact** [comment]
So we know for sure fox uses it to pick up chicks

**Comment by Fox-McCloud** [comment]
@optimumtact 

I rarely use the PDA, to be honest.

**Comment by paprka** [comment]
Erpers probably use the PDA to talk to each other and exchange erp info or whatever

Ghosts don't need to be seeing that, it helps keep ERP off the server and remove the people shitstorming over erpers and ghost training them when they try to keep it private. This is in fact the only way to do it without getting a trail of ghosts to cause a big deal, which was the entire problem in the first place.

**Comment by Ikarrus** [comment]
I approve of this just because it makes my job as admin easier.

**Comment by paprka** [comment]
can't admins already see pda messages

**Comment by Ikarrus** [comment]
No, we need to VV the PDA itself and hope the messages haven't been cleared. Otherwise, we'd need to lag the server and load up the logs.

Either way, we don't know whether or not someone is being a shit in PDA messages because we'd need to actively search for something to catch it.

**Comment by Ikarrus** [comment]
BTW, I'd like it if we either classified it under ghost whispers, or just always have it visible to ghosts.

I often disable ghost ears because of the fucking spam, and PDA messages aren't even all that common.

**Comment by Ikarrus** [comment]
No we need to VV the pda to read the messages. Or lag the server and load up the logs.

**Comment by TZK13** [comment]
As somebody that has probably snooped around PDA message logs the most out of anybody I don't really see any reason for this not to be a feature, PDA messaging simply isn't as private as people think it is. It's especially funny when you see antags being open about their plans on it.

>This is in fact the only way to do it without getting a trail of ghosts to cause a big deal, which was the entire problem in the first place.

Pen and paper is far more private in terms of nobody being able to read it IC if you do it properly, I don't even know if what's written on paper is logged.

**Comment by Mandurrrh** [comment]
I can change it to whisper if that makes it easier. As for the erpers, it's not allowed on the server and PDAs are not meant for ooc communications they are part of IC gameplay. So if they want to ERP they'll have to take it off the server. This allows you to follow traitors better, see the ai messing with anyone as Malf, allow you to see communications when an ai is one humaned, and just General round information to follow along as an observer. Looc is for explaining game mechanics not for following the round.

**Comment by TZK13** [comment]
A lot of PDA messaging is used for IC relationship stuff but ultimately ghosts can already see what you say to your IC partner in person.

**Comment by Mandurrrh** [comment]
HADES used to do this fun malf ai gimmick where he would send anonymous Pda messages to creep people out. If you didn't see the messages you missed this entire gimmick and play that was going on.

**Comment by paprka** [comment]
>No, we need to VV the PDA itself and hope the messages haven't been cleared. Otherwise, we'd need to lag the server and load up the logs.

This PR is fine then, if it adds functionality for admins investigating things while sacrificing people using PDAs for wiki links and stuff (which admins have already said are ok) then this is a good change.

**Comment by TZK13** [comment]
I don't see how this sacrifices wiki links being sent via PDA, ghosts seeing it doesn't make it more or less IC or anything.

**Comment by Fox-McCloud** [comment]
@Ikarrus 

Not so sure about that; on Para we have an investigate PDA button; it doesn't really incur any lag at all, and I don't believe people clearing their messages impacts it.

**Comment by Ikarrus** [comment]
Well, that goes to show that this isn't para station, I suppose. 

Still, it means admins can't survey in-game communication as it happens, and requires us to search the logs after the fact.

**Comment by Fox-McCloud** [comment]
@Ikarrus 

Wasn't meaning to imply it was; was just informing you the feature is out there and that it doesn't generate as much lag/issues as you speculate s'all.

**Comment by MrStonedOne** [comment]
@Fox-McCloud the lag he is taking about is the lag caused by sending somebody the log file.

@Ikarrus as far as my testings have shown, sending the log file generally only lags out the person getting it, maybe we should play around with this and see?

**Comment by Jordie0608** [comment]
http://keepachangelog.com/

**Comment by Mandurrrh** [comment]
IM SORRY JORDIE. I'm going to switch it to whisper anyways so I'll add my changelogs.

File: code/game/objects/items/devices/PDA/PDA.dm [hunk 1]
```diff
@@ -816,6 +816,9 @@ var/global/list/obj/item/device/pda/PDAs = list()
 
 		tnote += "<i><b>&rarr; To [P.owner]:</b></i><br>[t]<br>"
 		P.tnote += "<i><b>&larr; From <a href='byond://?src=\ref[P];choice=Message;target=\ref[src]'>[owner]</a> ([ownjob]):</b></i><br>[t]<br>"
+		for(var/mob/M in player_list)
+			if(M.stat == DEAD && M.client && (M.client.prefs.toggles & CHAT_GHOSTWHISPER))
```
Comment (Miauw62) [inline code/game/objects/items/devices/PDA/PDA.dm]: ```GHOSTWISPER```?

**Comment by Miauw62** [comment]
why is it whispers ;_;

**Comment by TheVekter** [comment]
I think the idea was so that we could hear all PDA messages even if we opt out of ghostears. I know some admins don't have ghostears on because it clogs up the text window too much.

**Comment by Miauw62** [comment]
just make it a separate toggle in that case.

**Comment by TheVekter** [comment]
Why the close?

**Comment by GunHog** [comment]
Git bends her over and does terrible things to her on occasion.

**Pull request merged**